examples/api/lib/material/text_button/text_button.0.dart

```diff
@@ -6,78 +6,417 @@ import 'package:flutter/material.dart';
 
 /// Flutter code sample for [TextButton].
 
-void main() => runApp(const TextButtonExampleApp());
+void main() {
+  runApp(const TextButtonExampleApp());
+}
+
+class TextButtonExampleApp extends StatefulWidget {
+  const TextButtonExampleApp({ super.key });
+
+  @override
+  State<TextButtonExampleApp> createState() => _TextButtonExampleAppState();
+}
 
-class TextButtonExampleApp extends StatelessWidget {
-  const TextButtonExampleApp({super.key});
+class _TextButtonExampleAppState extends State<TextButtonExampleApp> {
+  bool darkMode = false;
 
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      themeMode: darkMode ? ThemeMode.dark : ThemeMode.light,
+      theme: ThemeData(brightness: Brightness.light),
+      darkTheme: ThemeData(brightness: Brightness.dark),
       home: Scaffold(
-        appBar: AppBar(title: const Text('TextButton Sample')),
-        body: const TextButtonExample(),
+        body: Padding(
+          padding: const EdgeInsets.all(16),
+          child: TextButtonExample(
+            darkMode: darkMode,
+            updateDarkMode: (bool value) {
+              setState(() { darkMode = value; });
+            },
+          ),
+        ),
       ),
     );
   }
 }
 
-class TextButtonExample extends StatelessWidget {
-  const TextButtonExample({super.key});
+class TextButtonExample extends StatefulWidget {
+  const TextButtonExample({ super.key, required this.darkMode, required this.updateDarkMode });
+
+  final bool darkMode;
+  final ValueChanged<bool> updateDarkMode;
+
+  @override
+  State<TextButtonExample> createState() => _TextButtonExampleState();
+}
+
+class _TextButtonExampleState extends State<TextButtonExample> {
+  TextDirection textDirection = TextDirection.ltr;
+  ThemeMode themeMode = ThemeMode.light;
+  VisualDensity visualDensity = VisualDensity.standard;
+
+  static const Widget verticalSpacer = SizedBox(height: 16);
 
   @override
   Widget build(BuildContext context) {
-    return Center(
-      child: Column(
-        mainAxisSize: MainAxisSize.min,
-        children: <Widget>[
-          TextButton(
-            style: TextButton.styleFrom(
-              textStyle: const TextStyle(fontSize: 20),
+    final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
+
+    // Adapt colors that are not part of the color scheme to
+    // the current dark/light mode. Used to define TextButton #7's
+    // gradients.
+    final Color color1;
+    final Color color2;
+    final Color color3;
+    switch (colorScheme.brightness) {
+      case Brightness.light:
+        color1 = Colors.blue.withOpacity(0.5);
+        color2 = Colors.orange.withOpacity(0.5);
+        color3 = Colors.yellow.withOpacity(0.5);
+      case Brightness.dark:
+        color1 = Colors.purple.withOpacity(0.5);
+        color2 = Colors.cyan.withOpacity(0.5);
+        color3 = Colors.yellow.withOpacity(0.5);
+    }
```

Comment:
```suggestion
    final (Color color1, Color color2, Color color3) = switch (colorScheme.brightness) {
      Brightness.light => (
        Colors.blue.withOpacity(0.5),
        Colors.orange.withOpacity(0.5),
        Colors.yellow.withOpacity(0.5),
      ),
      Brightness.dark => (
        Colors.purple.withOpacity(0.5),
        Colors.cyan.withOpacity(0.5),
        Colors.yellow.withOpacity(0.5),
      ),
    };
```

Reply:
This is _so_ much better. It's hard (It was hard for me) to find an example of this syntax in action. Hopefully we're helping spread the word.